x-pack/plugins/ml/public/application/explorer/anomaly_context_menu.tsx

```diff
@@ -103,6 +136,183 @@ export const AnomalyContextMenu: FC<AnomalyContextMenuProps> = ({
+        maxSeriesToPlot: maxSeriesToPlot ?? DEFAULT_MAX_SERIES_TO_PLOT,
+        severityThreshold: severity.val,
+        ...((isDefined(queryString) && queryString !== '') ||
+        (queryFromSelectedCells !== undefined && queryFromSelectedCells !== '')
+          ? {
+              query: {
+                query: queryString === '' ? queryFromSelectedCells : queryString,
+                language: SEARCH_QUERY_LANGUAGE.KUERY,
+              } as Query,
+            }
+          : {}),
+      };
+    },
+    [jobIds, maxSeriesToPlot, severity, queryString, selectedCells]
+  );
+
+  const onSaveCallback: SaveModalDashboardProps['onSave'] = useCallback(
+    ({ dashboardId }) => {
+      const stateTransfer = embeddable!.getStateTransfer();
+
+      const embeddableInput: Partial<AnomalyChartsEmbeddableInput> = getEmbeddableInput();
+
+      const state = {
+        input: embeddableInput,
+        type: ANOMALY_EXPLORER_CHARTS_EMBEDDABLE_TYPE,
+      };
+
+      const path = dashboardId === 'new' ? '#/create' : `#/view/${dashboardId}`;
+
+      stateTransfer.navigateToWithEmbeddablePackage('dashboards', {
+        state,
+        path,
+      });
+    },
+    [embeddable, getEmbeddableInput]
+  );
+
+  const panels = useMemo<EuiContextMenuProps['panels']>(() => {
+    const rootItems: EuiContextMenuPanelItemDescriptor[] = [];
+    const menuPanels: EuiContextMenuPanelDescriptor[] = [{ id: 'panelActions', items: rootItems }];
+
+    const getContent = (callback: ReactEventHandler<HTMLButtonElement>) => (
+      <EuiPanel paddingSize={'s'}>
+        <EuiSpacer size={'s'} />
+        <EuiForm>
+          <EuiFormRow
+            isInvalid={!isMaxSeriesToPlotValid}
+            error={
+              !isMaxSeriesToPlotValid ? (
+                <FormattedMessage
+                  id="xpack.ml.anomalyChartsEmbeddable.maxSeriesToPlotError"
+                  defaultMessage="Maximum number of series to plot must be between 1 and 50."
+                />
+              ) : undefined
+            }
+            label={
+              <FormattedMessage
+                id="xpack.ml.explorer.addToDashboard.anomalyCharts.maxSeriesToPlotLabel"
+                defaultMessage="Maximum number of series to plot"
+              />
+            }
+          >
+            <EuiFieldNumber
+              data-test-subj="mlAnomalyChartsInitializerMaxSeries"
+              id="selectMaxSeriesToPlot"
+              name="selectMaxSeriesToPlot"
+              value={maxSeriesToPlot}
+              onChange={(e) => setMaxSeriesToPlot(parseInt(e.target.value, 10))}
+              min={1}
+              max={MAX_ANOMALY_CHARTS_ALLOWED}
+            />
+          </EuiFormRow>
+
+          <EuiButton
+            fill
+            type={'submit'}
+            fullWidth
+            onClick={callback}
+            disabled={!isMaxSeriesToPlotValid}
+          >
+            <FormattedMessage
+              id="xxpack.ml.explorer.anomalies.submitAttachLabel"
```

Comment:
See previous comment about prefixes for the ids in this file, but this one also starts with typo `xxpack`.

Reply:
It's been fixed